capstone/src/main/java/com/google/sps/Objects/response/MemberResponse.java

```diff
@@ -0,0 +1,96 @@
+package com.google.sps.Objects.response;
+
+import java.util.ArrayList;
+import com.google.sps.Objects.Badge;
+import java.util.LinkedHashSet;
+import com.google.appengine.api.datastore.Entity;
+
+/**
+ * Contains information for the response sent by GroupMembersServlet.
+ */
+public final class MemberResponse {
+
+  private final String profilePic;
+  private final String firstName;
+  private final String lastName;
+  private final LinkedHashSet<Badge> badges;
+  private final String userId;
+
+  /**
+   * Constructs a MemberResponse object.
+   *
+   * @param profilePic String url of profile image 
+   * @param firstName String of user first name 
+   * @param lastName String of user last name
+   * @param badges List of badge objects 
+   * @param userId user id 
+   */
+  public MemberResponse(String profilePic, String firstName, String lastName, LinkedHashSet<Badge> badges, String userId) {
+    this.profilePic = profilePic;
+    this.badges = badges;
+    this.firstName = firstName;
+    this.lastName = lastName;
+    this.userId = userId;
+  }
+  
+  /**
+   * Constructs a MemberResponse object from an entity
+   *
+   * @param profilePic String url of profile image 
+   * @param firstName String of user first name 
+   * @param lastName String of user last name  
+   * @param badges List of badge objects 
+   * @param userId user id 
+   */
+  public static MemberResponse fromEntity(Entity entity) {
```

Comment:
If you want to avoid adding badges in some scenarios, you could do the following:

```java
  public static MemberResponse fromEntity(Entity entity) {
    return MemberResponse.fromEntity(entity, /* includeBadges= */ true);
  }

  public static MemberResponse fromEntity(Entity entity, boolean includeBadges) {
    [Insert Your Current Logic]
    [check if includeBadges before doing anything with badges]
  }
```

Reply:
removed basicMember object and added includeBadges check